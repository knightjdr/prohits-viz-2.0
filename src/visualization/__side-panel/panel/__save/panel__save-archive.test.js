import React from 'react';
import { shallow } from 'enzyme';

import Archive from './panel__save-archive';

const archive = jest.fn();

describe('Archive session component', () => {
  beforeEach(() => {
    /* Clear call count */
    archive.mockClear();
  });

  it('should render', () => {
    const wrapper = shallow(
      <Archive
        archive={archive}
      />,
    );
    expect(wrapper).toMatchSnapshot();
  });

  it('should call archive prop method on button click', () => {
    const wrapper = shallow(
      <Archive
        archive={archive}
      />,
    );
    wrapper.find('Button').simulate('click');
    expect(archive).toHaveBeenCalled();
  });
});
