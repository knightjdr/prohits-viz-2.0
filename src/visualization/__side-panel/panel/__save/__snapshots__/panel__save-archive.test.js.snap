// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Archive session component should render 1`] = `
<div
  className="panel__save-archive"
>
  <div>
    <div>
      Archive
    </div>
    <Button
      className="panel__save-button"
      disabled={false}
      onClick={[Function]}
      type="default"
    >
      <FontAwesomeIcon$1
        border={false}
        className=""
        fixedWidth={false}
        flip={null}
        icon={
          Object {
            "icon": Array [
              512,
              512,
              Array [],
              "f187",
              "M32 448c0 17.7 14.3 32 32 32h384c17.7 0 32-14.3 32-32V160H32v288zm160-212c0-6.6 5.4-12 12-12h104c6.6 0 12 5.4 12 12v8c0 6.6-5.4 12-12 12H204c-6.6 0-12-5.4-12-12v-8zM480 32H32C14.3 32 0 46.3 0 64v48c0 8.8 7.2 16 16 16h480c8.8 0 16-7.2 16-16V64c0-17.7-14.3-32-32-32z",
            ],
            "iconName": "archive",
            "prefix": "fas",
          }
        }
        listItem={false}
        mask={null}
        name=""
        pull={null}
        pulse={false}
        rotation={null}
        size={null}
        spin={false}
        symbol={false}
        transform={null}
      />
    </Button>
  </div>
  <div>
    <FontAwesomeIcon$1
      border={false}
      className=""
      fixedWidth={false}
      flip={null}
      icon={
        Object {
          "icon": Array [
            512,
            512,
            Array [],
            "f06a",
            "M504 256c0 136.997-111.043 248-248 248S8 392.997 8 256C8 119.083 119.043 8 256 8s248 111.083 248 248zm-248 50c-25.405 0-46 20.595-46 46s20.595 46 46 46 46-20.595 46-46-20.595-46-46-46zm-43.673-165.346l7.418 136c.347 6.364 5.609 11.346 11.982 11.346h48.546c6.373 0 11.635-4.982 11.982-11.346l7.418-136c.375-6.874-5.098-12.654-11.982-12.654h-63.383c-6.884 0-12.356 5.78-11.981 12.654z",
          ],
          "iconName": "exclamation-circle",
          "prefix": "fas",
        }
      }
      listItem={false}
      mask={null}
      name=""
      pull={null}
      pulse={false}
      rotation={null}
      size={null}
      spin={false}
      symbol={false}
      transform={null}
    />
     
    <span>
      Archiving an image will generate a new URL that can be used to access your session from any browser. These sessions will be saved for three months. If you'd like to have your session permanently archived, please send us the URL and we will add it to the whitelist. See the
       
      <Component
        className="panel__save-link"
        to="/help/visualization/save/archive"
      >
        help
      </Component>
       
      for details.
    </span>
  </div>
</div>
`;
